sentry/src/main/java/io/sentry/IScopes.java

```diff
@@ -0,0 +1,594 @@
+package io.sentry;
+
+import io.sentry.metrics.MetricsApi;
+import io.sentry.protocol.SentryId;
+import io.sentry.protocol.SentryTransaction;
+import io.sentry.protocol.User;
+import io.sentry.transport.RateLimiter;
+import java.util.List;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface IScopes {
```

Comment:
Is there a specific reason we name it plural `IScopes` instead of `IScope`?

Reply:
`IScope` is already present (i.e. the scope) and `IScopes` is a set of multiple scopes (isolation scope and current scope) which (sometimes) fork together or separately. `IScopes` (and `Scopes`) is meant to be a drop-in replacement for hub.